ffi/src/log.rs

```diff
@@ -77,3 +77,60 @@ pub extern "C" fn set_logging_callback(
         }
     }
 }
+
+#[cfg(test)]
```

Comment:
i recently learned that there is an easy way to split out tests into separate file, e.g more like in golang. do you like that or not so much?

```rs
#[cfg(test)]
#[path = "log_test.rs"]
pub mod test;
```

and then in log_test.rs

```rs
 #[test]
fn test_c_logger() {
...
}
```

Reply:
Yeah, it is possible, but by convention in Rust, the unit tests go in the same file as the implementation (https://doc.rust-lang.org/book/ch11-03-test-organization.html). I think it makes sense to keep them close, especially if the tests are simple, there aren't many of them. If there are many/they are complex, it's an indication that maybe they should be integration tests (these land in separate _tests_ directory.